python/pylibcudf/pylibcudf/tests/test_hashing.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import hashlib
+import struct
+
+import mmh3
+import numpy as np
+import pyarrow as pa
+import pylibcudf as plc
+import pytest
+import xxhash
+from utils import assert_column_eq, assert_table_eq
+
+SEED = 0
+METHODS = ["md5", "sha1", "sha224", "sha256", "sha384", "sha512"]
+
+
+def scalar_to_binary(x):
+    if isinstance(x, str):
+        return x.encode()
+    elif isinstance(x, float):
+        return struct.pack("<d", x)
+    elif isinstance(x, bool):
+        return x.to_bytes(1, byteorder="little", signed=True)
+    elif isinstance(x, int):
+        return x.to_bytes(8, byteorder="little", signed=True)
+    else:
+        raise NotImplementedError
+
+
+def libcudf_mmh3_x86_32(binary):
+    seed = plc.hashing.LIBCUDF_DEFAULT_HASH_SEED
+    hashval = mmh3.hash(binary, seed)
+    return seed ^ (hashval + 0x9E3779B9 + (seed << 6) + (seed >> 2))
```

Comment:
Where is this coming from? These values don't match the [libcudf implementation of murmurhash32_x86_32](https://github.com/rapidsai/cudf/blob/4fe338c0efe0fee2ee69c8207f9f4cbe9aa4d4a2/cpp/include/cudf/hashing/detail/murmurhash3_x86_32.cuh).

Reply:
Ah, I see. We should separate this function into two. The first part should call `mmh3`. The second part (the last line here) should be a standalone helper function called `hash_combine_32` to mirror https://github.com/rapidsai/cudf/blob/4fe338c0efe0fee2ee69c8207f9f4cbe9aa4d4a2/cpp/include/cudf/hashing/detail/hashing.hpp#L85-L88.